common/src/main/java/io/netty/util/concurrent/PromiseCombiner.java

```diff
@@ -36,23 +50,44 @@ public void operationComplete(Future<?> future) throws Exception {
         }
     };
 
+    /**
+     * Adds a new future to be combined. New futures may be added until an aggregate promise is added via the
+     * {@link PromiseCombiner#finish(Promise)} method has been called.
+     *
+     * @param future the future to add to this promise combiner
+     */
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    public void add(Promise promise) {
+    public void add(final Future future) {
         checkAddAllowed();
         ++expectedCount;
-        promise.addListener(listener);
+        future.addListener(listener);
     }
 
+    /**
+     * Adds new futures to be combined. New futures may be added until an aggregate promise is added via the
+     * {@link PromiseCombiner#finish(Promise)} method has been called.
+     *
+     * @param futures the futures to add to this promise combiner
+     */
     @SuppressWarnings({ "unchecked", "rawtypes" })
-    public void addAll(Promise... promises) {
-        checkAddAllowed();
-        expectedCount += promises.length;
-        for (Promise promise : promises) {
-            promise.addListener(listener);
+    public void addAll(final Future... futures) {
```

Comment:
@jchambers we can not make this change as it would break this:

```
Promise[] promises = ...;
combiner.addAll(promises);
```

Please keep the old method and mark it as `@deprecated`